modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterManagementGroupManager.java

```diff
@@ -389,45 +388,44 @@ private void onElectedAsLeader(long term) {
                     }
                 });
 
-        raftServiceAfterJoin().thenCompose(this::pushClusterConfigToCluster);
-    }
-
-    private CompletableFuture<Void> pushClusterConfigToCluster(CmgRaftService service) {
-        return service.readClusterState()
-                .thenCompose(state -> {
-                    if (state == null) {
-                        LOG.info("No CMG state found in the Raft service");
-                        return completedFuture(null);
-                    } else if (state.clusterConfigurationToApply() == null) {
-                        // Config was applied or wasn't provided
-                        LOG.info("No cluster configuration found in the Raft service");
-                        return completedFuture(null);
-                    } else {
-                        LOG.info("Cluster configuration is found in the Raft service, going to apply it");
-                        return distributedConfigurationUpdater.updateConfiguration(state.clusterConfigurationToApply())
-                                .thenCompose(unused -> removeClusterConfigFromClusterState(service));
-                    }
-                });
+        raftServiceAfterJoin().whenComplete((service, e) -> {
+            if (e != null) {
+                LOG.error("Error when joining to the raft service", e);
+                updateDistributedConfigurationActionFuture.completeExceptionally(e);
+            } else {
+                service.readClusterState()
+                        .thenAccept(state -> {
```

Comment:
I'll copy my previous comment here: I think we need to use `whenComplete` here to complete `updateDistributedConfigurationActionFuture` with exceptions in case previous operations fail